sydent/http/servlets/__init__.py

```diff
@@ -163,45 +163,27 @@ def inner(self, request: Request, *args, **kwargs) -> bytes:
 
 
 def asyncjsonwrap(f):
-    def reqDone(resp: Dict[str, Any], request: Request) -> None:
-        """
-        Converts the given response content into JSON and encodes it to bytes, then
-        writes it as the response to the given request with the right headers.
-
-        :param resp: The response content to convert to JSON and encode.
-        :param request: The request to respond to.
-        """
-        request.setHeader("Content-Type", "application/json")
-        request.write(dict_to_json_bytes(resp))
-        request.finish()
-
-    def reqErr(failure: Failure, request: Request) -> None:
-        """
-        Logs the given failure. If the failure is a MatrixRestError, writes a response
-        using the info it contains, otherwise responds with 500 Internal Server Error.
-
-        :param failure: The failure to process.
-        :param request: The request to respond to.
-        """
+    async def render(f, servlet, request, **kwargs):
         request.setHeader("Content-Type", "application/json")
-        if failure.check(MatrixRestError) is not None:
-            request.setResponseCode(failure.value.httpStatus)
+        try:
+            result = await f(servlet, request, **kwargs)
+            request.write(dict_to_json_bytes(result))
+            request.finish()
+        except MatrixRestError as e:
+            request.setResponseCode(e.value.httpStatus)
```

Comment:
The `MatrixRestError` has the `httpStatus` directly, previously `f.value` gave the `MatrixRestError` instance.

```suggestion
            request.setResponseCode(e.httpStatus)
```

Reply:
Thanks for explaining that I didn't quite catch that last time around :)